### ISSUE TITLE
issue with s3 and new bucketname

### ISSUE THREAD

**Issue by hamedhsn** [pinterest/secor#120]
I have setup secor but when I run test_run, i get the following error. note that I was successfully able to set it up with another bucket and it works find for that. but when I change the s3 bucket to a different name(I have also created the bucket manually) it provides such error. 

See log /tmp/secor_dev/logs/log_verifier_backup.log for more details
2015-08-14 14:06:46,037 [main] (com.pinterest.secor.main.LogFileVerifierMain:99) ERROR Log file verifier failed
org.apache.hadoop.fs.s3.S3Exception: org.jets3t.service.S3ServiceException: S3 HEAD request failed for '/sector-test%2Fbackup%2Ftest' - ResponseCode=400, ResponseMessage=Bad Request
	at org.apache.hadoop.fs.s3native.Jets3tNativeFileSystemStore.handleServiceException(Jets3tNativeFileSystemStore.java:229)
	at org.apache.hadoop.fs.s3native.Jets3tNativeFileSystemStore.retrieveMetadata(Jets3tNativeFileSystemStore.java:111)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:85)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:62)
	at org.apache.hadoop.fs.s3native.$Proxy1.retrieveMetadata(Unknown Source)
	at org.apache.hadoop.fs.s3native.NativeS3FileSystem.listStatus(NativeS3FileSystem.java:407)
	at com.pinterest.secor.util.FileUtil.list(FileUtil.java:55)
	at com.pinterest.secor.util.FileUtil.listRecursively(FileUtil.java:71)
	at com.pinterest.secor.tools.LogFileVerifier.populateTopicPartitionToOffsetToFiles(LogFileVerifier.java:62)
	at com.pinterest.secor.tools.LogFileVerifier.verifyCounts(LogFileVerifier.java:120)
	at com.pinterest.secor.main.LogFileVerifierMain.main(LogFileVerifierMain.java:93)

**Comment by hamedhsn** [comment]
After spending two hours and weird message at the end found that the bucket that I was creating was in frankfurt region. I can not find any reason why writing fails in that region. Anyway that was the issue!!

**Comment by ppearcy** [comment]
Had same problem.

The key problem here is that HMAC v2 is being used and eu-central-1 requires v4. Looks like there is work needed in HadoopS3UploadManager to support this. Needs to switch to s3a and add option to set the endpoint so jets3t can do the right thing.  

I was able to get eu-central working using S3UploadManager, though. I needed to exclude the aws sdk brought in by hadoop-aws while compiling since it causes various classloader issues with the latest aws sdk. Secor's jar deps is sort of becoming a mess as more and more is getting pushed into the core app vs a plugin approach. 

Then I needed to ensure that endpoint or region is set and things work.
